src/view/ui.py

```diff
@@ -97,7 +138,29 @@ def initUI(self):
         self.proxy = QtWidgets.QGraphicsProxyWidget(self.handle)
         self.proxy.setWidget(self)
 
+    def setWidget(self, widget):
+        """
+        update the central widget with new widget
+
+        Parameters
+        ----------
+        widget: QWidget
+        """
+        for k, v in widget.__dict__.items():
+            if k in self.__dict__:
+                return print("setWidget FAILED: you cannot set widget in \
+                    QViewWidget that contains a parameter named '{}'".format(k))
```

Comment:
You use a for loop over items(), but it seems you do not use `v` inside the loop.
If I understand, you want to compare the keys of `widget` and those of your object?

Reply:
yes because I merge the attributes of both widget and object to simplify their use inside graph. But I need to check if attributes does not already exist inside QViewWidget..